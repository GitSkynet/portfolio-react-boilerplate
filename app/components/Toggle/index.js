/**
*
* LocaleToggle
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import enFlag from './en.png';
import nlFlag from './nl.png';
import Wrapper from './Wrapper';

import ToggleOption from '../ToggleOption';

const imagesLookup = {
  en: enFlag,
  nl: nlFlag,
};

function Toggle({ currentValue, messages, onToggle, values }) {
  let content = (<div />);

  // If we have items, render them
  if (values) {
    content = values.map((value) => (
      <ToggleOption
        key={value}
        value={value}
        isChecked={value === currentValue}
        message={messages[value]}
        onChange={onToggle}
        iconImg={imagesLookup[value]}
      />
    ));
  }

  return (
    <Wrapper>
      {content}
    </Wrapper>
  );
}

Toggle.propTypes = {
  onToggle: PropTypes.func,
  values: PropTypes.array,
  currentValue: PropTypes.string,
  messages: PropTypes.object,
};

export default Toggle;
