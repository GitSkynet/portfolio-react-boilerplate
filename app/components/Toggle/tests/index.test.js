import React from 'react';
import { shallow } from 'enzyme';
import { IntlProvider, defineMessages } from 'react-intl';

import Toggle from '../index';

describe('<Toggle />', () => {
  it('should contain default text', () => {
    const defaultEnMessage = 'someContent';
    const defaultNlMessage = 'someOtherContent';
    const messages = defineMessages({
      en: {
        id: 'boilerplate.containers.LocaleToggle.en',
        defaultMessage: defaultEnMessage,
      },
      nl: {
        id: 'boilerplate.containers.LocaleToggle.en',
        defaultMessage: defaultNlMessage,
      },
    });
    const renderedComponent = shallow(
      <IntlProvider locale="en">
        <Toggle values={['en', 'nl']} messages={messages} />
      </IntlProvider>
    );

    expect(renderedComponent.contains(<Toggle values={['en', 'nl']} messages={messages} />)).toBe(true);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should not have ToggleOptions if props.values is not defined', () => {
    const renderedComponent = shallow(<Toggle />);
    expect(renderedComponent.contains(<div />)).toBe(true);
    expect(renderedComponent.find('div').length).toBe(1);
  });
});
