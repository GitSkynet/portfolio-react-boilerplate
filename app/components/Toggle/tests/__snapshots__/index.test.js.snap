// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should contain default text 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider
    locale="en"
>
    <Toggle
        messages={
            Object {
                "en": Object {
                  "defaultMessage": "someContent",
                  "id": "boilerplate.containers.LocaleToggle.en",
                },
                "nl": Object {
                  "defaultMessage": "someOtherContent",
                  "id": "boilerplate.containers.LocaleToggle.en",
                },
              }
        }
        values={
            Array [
                "en",
                "nl",
              ]
        }
    />
</IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "messages": Object {
        "en": Object {
          "defaultMessage": "someContent",
          "id": "boilerplate.containers.LocaleToggle.en",
        },
        "nl": Object {
          "defaultMessage": "someOtherContent",
          "id": "boilerplate.containers.LocaleToggle.en",
        },
      },
      "values": Array [
        "en",
        "nl",
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "messages": Object {
          "en": Object {
            "defaultMessage": "someContent",
            "id": "boilerplate.containers.LocaleToggle.en",
          },
          "nl": Object {
            "defaultMessage": "someOtherContent",
            "id": "boilerplate.containers.LocaleToggle.en",
          },
        },
        "values": Array [
          "en",
          "nl",
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
